R/utilities.R

```diff
@@ -0,0 +1,540 @@
+library(tidyverse)
+library(behindbarstools)
+library(docstring)
+
+# ------------------------------------------------------------------------------
+# READ SHEETS 
+# ------------------------------------------------------------------------------
+
+read_new_fac_info <- function(google_sheet_url = NULL) {
+    #' Read in UCLA facility data updates 
+    #'
+    #' Reads in new entries for the facility info sheet from Google Sheets.  
+    #'
+    #' @param google_sheet_url character string URL for the Google Sheet 
+    #'
+    #' @return data frame 
+    
+    if (is.null(google_sheet_url)) {
+        google_sheet_url <- "1tAhD-brnuqw0s55QXM-xYLPsyD-rNrqHbAVIbxSOMwI"
+    }
+    
+    google_sheet_url %>% 
+        googlesheets4::read_sheet(sheet = "fac_data", 
+                                  range = readxl::cell_cols("A:W"), 
+                                  col_types = "cccccccdcddddcccccddccc") %>% 
+        mutate(Name = clean_fac_col_txt(Name, to_upper = TRUE)) %>% 
+        unique()
+}
+
+read_new_fac_spellings <- function(google_sheet_url = NULL) {
+    #' Read in UCLA facility spellings updates 
+    #'
+    #' Reads in new entries for the facility alternative spellings sheet from Google Sheets.  
+    #'
+    #' @param google_sheet_url character string URL for the Google Sheet 
+    #'
+    #' @return data frame 
+    
+    if (is.null(google_sheet_url)) {
+        google_sheet_url <- "1tAhD-brnuqw0s55QXM-xYLPsyD-rNrqHbAVIbxSOMwI"
+    }
+    
+    google_sheet_url %>% 
+        googlesheets4::read_sheet(sheet = "fac_spellings") %>% 
+        mutate(xwalk_name_raw = clean_fac_col_txt(xwalk_name_raw, to_upper = TRUE), 
+               xwalk_name_clean = clean_fac_col_txt(xwalk_name_clean, to_upper = TRUE)) %>% 
+        unique()
+}
+
+# ------------------------------------------------------------------------------
+# VALIDATE DATA 
+# ------------------------------------------------------------------------------
+
+is_valid_jurisdiction <- function(jurisdiction) {
+    #' Returns TRUE if the given jurisdiction is valid 
+    #' 
+    #' Returns TRUE if the given jurisdiction is in the list of valid 
+    #' jurisdictions (state, county, federal). 
+    #' 
+    #' @param jurisdiction character string of the jurisdiction to check 
+    #' 
+    #' @return logical, TRUE if the given state is a valid jurisdiction 
+    
+    valid_jurisdictions <- c(
+        "state", 
+        "county", 
+        "federal") 
```

Comment:
@nmmarquez - just realized that I'll have to add immigration as a jurisdiction here. You called it `immigration`, right? Or `ice`?

Reply:
yup yup lets go with immigration